server/utils/activityHooks.ts

```diff
@@ -0,0 +1,34 @@
+import { defer } from 'server/utils/deferred';
+
+type CreateActivityHooksOptions<Model> = {
+	Model: any;
+	onModelCreated?: (actorId: null | string, modelId: string) => Promise<void>;
+	onModelUpdated?: (
+		actorId: null | string,
+		modelId: string,
+		previousModel: Model,
+	) => Promise<void>;
+	onModelDestroyed?: (actorId: null | string, modelId: string) => Promise<void>;
+};
+
+export const createActivityHooks = <Model>(options: CreateActivityHooksOptions<Model>) => {
+	const { Model, onModelCreated, onModelUpdated, onModelDestroyed } = options;
```

Comment:
Super minor and feel free to ignore, but the generic parameter `Model` and constant of the same name `Model` reduces the readability of this function just a bit. Could potentially make the option lowercase?

Reply:
I went back and forth on casing here but I didn't even realize that I was using the same name twice here, so yeah, it should totally be lowercase.